Moq.AutoMock.Tests/DescribeAsDisposable.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Linq;
+
+namespace Moq.AutoMock.Tests
+{
+    [TestClass]
+    public class DescribeAsDisposable
+    {
+        [TestMethod]
+        public void It_disposes_all_registered_disposable_instances()
+        {
+            TestDisposable test = new();
+            AutoMocker mocker = new();
+            mocker.Use(test);
+            IDisposable disposable = mocker.AsDisposable();
+
+            disposable.Dispose();
+
+            Assert.IsTrue(test.IsDisposed);
+        }
+
+        [TestMethod]
+        public void It_disposes_using_custom_disposable_instance()
+        {
+            TestDisposable test = new();
+            AutoMocker mocker = new();
+            CustomDisposer disposer = new(mocker);
+            mocker.Use(test);
+            mocker.Use<IAutoMockerDisposable>(disposer);
+            IDisposable disposable = mocker.AsDisposable(); 
+
+            disposable.Dispose();
+
+            Assert.IsTrue(disposer is CustomDisposer);
```

Comment:
This can't be false.

Reply:
O good catch, it was supposed to be `disposable` not `disposer`

Reply:
Good catch, was supposed to be `disposable`